lib/routemaster/responses/future_response.rb

```diff
@@ -1,29 +1,16 @@
-require 'thread/pool'
-require 'thread/future'
+require 'concurrent/future'
+require 'concurrent/executor/cached_thread_pool'
 require 'singleton'
 require 'delegate'
 
 module Routemaster
   module Responses
-    # A pool of threads, used for parallel/future request processing.
-    class Pool < SimpleDelegator
-      include Singleton
-
-      def initialize
-        Thread.pool(5, 20).tap do |p|
-          p.auto_trim!
-          p.idle_trim! 10 # 10 seconds
-          super p
-        end
-      end
-    end
-
     class FutureResponse
       extend Forwardable
 
       # The `block` is expected to return a {Response}
       def initialize(&block)
-        @future = Pool.instance.future(&block)
+        @future = Concurrent::Future.execute(executor: Pool.current, &block)
```

Comment:
In the LD we're wrapping the current `thread` futures in a `delay`. This allows us to have a promise style `on_complete`/`then` block (we do some standard logging/error checking in there.

If we're upgrading anyway - could we use a `Promise`  here rathe than a `Future`?

https://ruby-concurrency.github.io/concurrent-ruby/Concurrent/Promise.html